src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -68,7 +79,12 @@ public CreateRequestService(CreateRequestRepositories repositories,
       .thenComposeAsync(r -> r.after(updateUponRequest.updateLoan::onRequestCreateOrUpdate))
       .thenComposeAsync(r -> r.after(requestRepository::create))
       .thenComposeAsync(r -> r.after(updateUponRequest.updateRequestQueue::onCreate))
-      .thenApply(r -> r.next(requestNoticeSender::sendNoticeOnRequestCreated));
+      .thenApplyAsync(r -> {
+        ofNullable(eventPublisher).ifPresent(publisher ->
```

Comment:
Can we use the existing request, instead of re-read it from storage?

Reply:
@bohdan-suprun looks like no because we need an information about request position that isn't available on existing request.

Reply:
What information is needed about the request position that isn't on the existing request?

Reply:
Request position in the queue. This call is in the separate async process and doesn't affect main flow.

Reply:
I don't understand, I thought each request had a property for it's position in the queue.

Reply:
Ah, yes, thank you both. Position is available without calling storage. I'll fix this.